exchanges/okx/okx.go

```diff
@@ -3701,82 +4942,77 @@ func (ok *Okx) CurrencyUnitConvert(ctx context.Context, instrumentID string, qua
 	}
-	var resp []UnitConvertResponse
-	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, unitConvertEPL, http.MethodGet, common.EncodeURLValues(publicCurrencyConvertContract, params), nil, &resp, false)
-	if err != nil {
-		return nil, err
+	var resp *UnitConvertResponse
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, unitConvertEPL, http.MethodGet, common.EncodeURLValues("public/convert-contract-coin", params), nil, &resp, false)
+}
+
+// GetOptionsTickBands retrieves option tick bands information.
+// Instrument type OPTION
+func (ok *Okx) GetOptionsTickBands(ctx context.Context, instrumentType, instrumentFamily string) ([]OptionTickBand, error) {
+	if instrumentType == "" {
+		return nil, errInstrumentTypeRequired
 	}
-	if len(resp) == 1 {
-		return &resp[0], nil
+	params := url.Values{}
+	params.Set("instType", instrumentType)
+	if instrumentFamily != "" {
+		params.Set("instFamily", instrumentFamily)
 	}
-	return nil, errNoValidResponseFromServer
+	var resp []OptionTickBand
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, optionTickBandsEPL, http.MethodGet, common.EncodeURLValues("public/instrument-tick-bands", params), nil, &resp, false)
 }
 
 // Trading Data Endpoints
 
 // GetSupportCoins retrieves the currencies supported by the trading data endpoints
 func (ok *Okx) GetSupportCoins(ctx context.Context) (*SupportedCoinsData, error) {
-	var response SupportedCoinsData
-	return &response, ok.SendHTTPRequest(ctx, exchange.RestSpot, getSupportCoinEPL, http.MethodGet, tradingDataSupportedCoins, nil, &response, false)
+	var response *SupportedCoinsData
+	return response, ok.SendHTTPRequest(ctx, exchange.RestSpot, getSupportCoinEPL, http.MethodGet, "rubik/stat/trading-data/support-coin", nil, &response, false, true)
 }
 
 // GetTakerVolume retrieves the taker volume for both buyers and sellers.
-func (ok *Okx) GetTakerVolume(ctx context.Context, currency, instrumentType string, begin, end time.Time, period kline.Interval) ([]TakerVolume, error) {
-	params := url.Values{}
+func (ok *Okx) GetTakerVolume(ctx context.Context, ccy currency.Code, instrumentType string, begin, end time.Time, period kline.Interval) ([]TakerVolume, error) {
 	if instrumentType == "" {
 		return nil, errInstrumentTypeRequired
 	}
+	params := url.Values{}
 	params.Set("instType", strings.ToUpper(instrumentType))
 	interval := ok.GetIntervalEnum(period, false)
 	if interval != "" {
 		params.Set("period", interval)
 	}
-	if currency != "" {
-		params.Set("ccy", currency)
+	if !ccy.IsEmpty() {
+		params.Set("ccy", ccy.String())
 	}
 	if !begin.IsZero() {
 		params.Set("begin", strconv.FormatInt(begin.UnixMilli(), 10))
 	}
 	if !end.IsZero() {
 		params.Set("end", strconv.FormatInt(end.UnixMilli(), 10))
 	}
-	var response [][3]string
-	err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getTakerVolumeEPL, http.MethodGet, common.EncodeURLValues(tradingTakerVolume, params), nil, &response, false)
+	var response [][3]types.Number
```

Comment:
For all these fixed sizes across this package you can just have a custom unmarshaller; saves another iteration and reduces code footprint a little bit.

```diff
diff --git a/exchanges/okx/okx.go b/exchanges/okx/okx.go
index 2fad6c63f..db0340eb7 100644
--- a/exchanges/okx/okx.go
+++ b/exchanges/okx/okx.go
@@ -4992,20 +4992,9 @@ func (ok *Okx) GetTakerVolume(ctx context.Context, ccy currency.Code, instrument
        if !end.IsZero() {
                params.Set("end", strconv.FormatInt(end.UnixMilli(), 10))
        }
-       var response [][3]types.Number
+       var response []TakerVolume
        err := ok.SendHTTPRequest(ctx, exchange.RestSpot, getTakerVolumeEPL, http.MethodGet, common.EncodeURLValues("rubik/stat/taker-volume", params), nil, &response, false)
-       if err != nil {
-               return nil, err
-       }
-       takerVolumes := make([]TakerVolume, len(response))
-       for x := range response {
-               takerVolumes[x] = TakerVolume{
-                       Timestamp:  time.UnixMilli(response[x][0].Int64()),
-                       SellVolume: response[x][1].Float64(),
-                       BuyVolume:  response[x][2].Float64(),
-               }
-       }
-       return takerVolumes, nil
+       return response, err
 }

 // GetMarginLendingRatio retrieves the ratio of cumulative amount between currency margin quote currency and base currency.
diff --git a/exchanges/okx/okx_types.go b/exchanges/okx/okx_types.go
index a3c43078c..ac6fdcb0f 100644
--- a/exchanges/okx/okx_types.go
+++ b/exchanges/okx/okx_types.go
@@ -548,9 +548,14 @@ type SupportedCoinsData struct {

 // TakerVolume represents taker volume information with creation timestamp
 type TakerVolume struct {
-       Timestamp  time.Time `json:"ts"`
-       SellVolume float64
-       BuyVolume  float64
+       Timestamp  convert.ExchangeTime
+       SellVolume types.Number
+       BuyVolume  types.Number
+}
+
+func (t *TakerVolume) UnmarshalJSON(data []byte) error {
+       deploy := [3]any{&t.Timestamp, &t.SellVolume, &t.BuyVolume}
+       return json.Unmarshal(data, &deploy)
 }
```